tests/test_postgres_logical_replication.py

```diff
@@ -555,10 +637,137 @@ def test_run(self):
+
+        #first record will be the previous delete
+        delete_message = records_by_stream['postgres_logical_replication_test']['messages'][0]
+        sdc_deleted_at = delete_message['data'].get('_sdc_deleted_at')
+        self.assertIsNotNone(sdc_deleted_at)
+        self.assertEqual(delete_message['data']['id'], 3)
+
+
+
+        #the 2nd message will be the more recent delete
+        delete_message = records_by_stream['postgres_logical_replication_test']['messages'][1]
+        self.assertEqual(delete_message['action'], 'upsert')
+
+        sdc_deleted_at = delete_message['data'].get('_sdc_deleted_at')
+        self.assertIsNotNone(sdc_deleted_at)
+        self.assertEqual(delete_message['data']['id'], 2)
+        print("deleted record is correct")
+
+        state = menagerie.get_state(conn_id)
+        bookmark = state['bookmarks']['dev-public-postgres_logical_replication_test']
+        self.assertIsNone(state['currently_syncing'], msg="expected state's currently_syncing to be None")
+
+        self.assertIsNotNone(bookmark['lsn'],
+                             msg="expected bookmark for stream ROOT-CHICKEN to have an scn")
+
+        lsn_4 = bookmark['lsn']
+        self.assertTrue(lsn_4 >= lsn_3)
+
+
+        #table_version does NOT change
+        self.assertEqual(bookmark['version'], table_version,
+                         msg="expected bookmark for stream postgres_logical_replication_test to match version")
+        #----------------------------------------------------------------------
+        # invoke the sync job again after deleting a record using the 'id IN (<id>, <id>)' format
+        #----------------------------------------------------------------------
+        print("delete row from source db")
+        with db_utils.get_test_connection('dev') as conn:
+            with conn.cursor(cursor_factory=psycopg2.extras.DictCursor) as cur:
+                cur.execute("DELETE FROM {} WHERE id IN (4, 5)".format(canonicalized_table_name(test_schema_name, test_table_name, cur)))
+
+        sync_job_name = runner.run_sync_mode(self, conn_id)
+
+        # verify tap and target exit codes
+        exit_status = menagerie.get_exit_status(conn_id, sync_job_name)
+        menagerie.verify_sync_exit_status(self, exit_status, sync_job_name)
+
+        record_count_by_stream = runner.examine_target_output_file(self,
+                                                                   conn_id,
+                                                                   self.expected_sync_streams(),
+                                                                   self.expected_pks())
+
+
+        self.assertEqual(record_count_by_stream, { 'postgres_logical_replication_test': 3 })
+        records_by_stream = runner.get_records_from_target_output()
+
+        for stream, recs in records_by_stream.items():
+            # verify the persisted schema was correct
+            self.assertEqual(recs['schema'],
```

Comment:
Should this be `self.assertDictEqual`?

Reply:
Maybe it can be, it seems like `self.assertEqual` does the same thing though. It seems to compare all key-value pairs independently regardless of ordering.